src/components/shared/ContractFunction.vue

```diff
@@ -170,14 +170,16 @@ function handleCall(e: Event) {
   }
 }
 
-const makeBytes32 = (value: string, type: string) => {
-  if (/^bytes32/.test(type)) {
+const makeBytes = (value: string, type: string) => {
+  if (/^bytes/.test(type)) {
+    const _bytesCount = type.match(/\d+/) ?? []
+    const bytesCount = _bytesCount.length > 0 ? Number(_bytesCount[0]) : 0
     if (/^[0-9]+$/.test(value)) {
       const hex = numberToHex(value)
-      return padLeft(hex, 64)
+      return padLeft(hex, bytesCount * 2)
     }
     if (/^0x[0-9a-f]*$/i.test(value)) {
-      return padLeft(value, 64)
+      return padRight(value, bytesCount * 2)
```

Comment:
Just for curiosity I know that @richtera mentioned it should be padRight instead but wondering if this change would be outputting a completely different result.. because padLeft would put all zeros to the left while the 2nd will do on the right. so just wondering of the impact of this change 🤔

Reply:
In most of the cases when we have raw bytes (e.g. a string, an address, anything else) we also encode and specify it's length before the actual value. Here we don't have the full picture but `padRight` is the correct way for raw bytes. 
It's part of the ABI encoding rules for everything on solidity.
If you encode 2 bytes string with ABI type `bytes` it will be:
- length `0x000000000000000000000000000000000000000000000002`
- value `0xcafe00000000000000000000000000000000000000000000`


```
0x000000000000000000000000000000000000000000000002cafe00000000000000000000000000000000000000000000
```

Here's a link to the specification: https://docs.soliditylang.org/en/develop/abi-spec.html#:~:text=enc(X)%20%3D%20enc(k)%20pad_right(X)

In the next point in the list you can also find that `string` type is encoded as UTF8 and is represented as `bytes` eventually so all the same rules apply.

This change must be a bug fix. If something breaks after that it's likely that what broke wasn't implemented right.

Reply:
Ok cool I am just double checking this. So it means that this code was always broken is that it?

Reply:
@JeneaVranceanu Yea this is not quite correct many times. It just required people to mess with strings and bytes differently. If you want to use padRight, then please only do it if the type == "string" or "bytes" rather than "bytes32". So for "uint256" or "bytesN" it needs to be left padded and for bytes and strings it could be right padded.

Reply:
Okay, so the [tokenID of LSP-8](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-8-IdentifiableDigitalAsset.md#:~:text=value%20is%20different.-,LSP8TokenIdFormat,-Left%20padded) needs some special treatment. I don't think every other argument that is `bytesN` needs to be left padded. For example, `bytes32` values or keys for `setData` should be right padded. That's my understanding.

Reply:
Looking at this again so uintN, intN and address should be left padded, bytesN and so on should be right padded. i.e. address is dealt with as a number.

Reply:
@JeneaVranceanu I rebased this and simplified the function lookup. The problem was that the cache key was the same as the URL so that it will cache and return the original request making it workin debugging but not during deployment. I also inserted the bytesN exception. Let me know if you can look at it to accept it.